DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -119,5 +156,31 @@ public void UpdateProgress(int progressId)
                 );
             }
         }
+
+        private bool DoesProgressExist(int candidateId, int customisationId)
+        {
+            return connection.Query<int>(
+                @"SELECT ProgressId
+                        FROM Progress
+                        WHERE CandidateID = @candidateId
+                          AND CustomisationID = @customisationId
+                          AND SystemRefreshed = 0
+                          AND RemovedDate IS NULL",
+                new { candidateId, customisationId }
+            ).Any();
+        }
+
+        private int GetProgressId(int candidateId, int customisationId)
+        {
+            return connection.QueryFirst<int>(

```

Comment:
Won't this throw an exception if there's no progress id?

Reply:
The method is currently only called after DoesProgressExist but if we change it to return int? like suggested above then I can put it in a try catch.